### ISSUE TITLE
Enable legacy support for VB templates

### ISSUE THREAD

**Issue by mrlacey** [Microsoft/WindowsTemplateStudio#1325]
The `BuildRightClickWithLegacyTests` depend on previous versions of templates existing. Because these don't exist for the VB templates the tests report false negatives.

These tests can't be run for VB projects until after the templates exist. 
Doing this is not possible until after the 1.7 milestone release.

There is a TODO note in `BuildRightClickWithLegacyFixture.cs` to address this.

Assigned to 2.0 but needs to be done for whatever comes after 1.7 (when VB support is officially released)

**Comment by mrlacey** [comment]
The current Legacy/RightClick tests just test using the latest templates to do a right-click>add to a project built with the last public release of the templates.  
This is the equivalent of checking the current templates work with a project built with version N-1.

In reality, I think we should be testing that the current templates work with projects built with _all_ previous versions.  Thoughts about this? - @sibille @crutkas @ralarcon

**Comment by ralarcon** [comment]
@mrlacey what we see from our telemetry is that most of the people using WTS belong to versions N and N-1.

There are a few people that get stuck in previous versions but probably they are not actually using WTS (that is the case of people using 1.2 or less, in fact, the 112 users from 1.2 version we are almost sure that there are inactive users as in that version the telemetry was initialized before the wizard was launched).

Following is the rough data from our telemetry regarding versions used:

WtsWizardFileVersion | Total | %   Total
-- | -- | --
1.7.18030.1 | 1.7   K | 85.0%
1.2.17198.1 | 172 | 8.6%
1.6.17355.2 | 114 | 5.7%
1.0.17142.1 | 7 | 0.3%
1.5.17334.7 | 7 | 0.3%
0.7.17240.1 | 1 | 0.0%

Then, the distribution of new projects vs right click is as follows:

WtsWizardType | Total | %   Total
-- | -- | --
NewProject | 158 | 88.8%
AddPage | 15 | 8.4%
AddFeature | 5 | 2.8%

I agree that testing that our current templates against projects generated with all previous is interesting but will require more time for builds to complete and considering the numbers I'm not sure if it will be worthy.

**Comment by mrlacey** [comment]
I don't see how that data is relevant to the issue here.  
The data that's needed is, when someone adds a page or a feature, what version of the templates was used to generate the original project?  
Is this data captured?

Without any data, my expectation is that people will add to originally generated projects over time. There may be some who add shortly after a project is created but I would expect that the right-click>add functionality is more appropriate to apps that are updated in the future and when a newer version of templates is available. 

I'll update the existing tests to run the same way they were before but I think we should have tests that verify adding new pages or features to apps that were generated with any previous version of the templates.

This will really matter when breaking changes are introduced but let's hope that checks over a single version difference catch these.

**Comment by ralarcon** [comment]
The most relevant thing for me is that around 10% of the usage become from right-clicking. That said, I think is interesting to have as many tests as possible. But at the same time, building all possible combinations framework and project types with all pages for all previous versions will be a bit time-consuming. Instead of removing it, why don't we set those tests in a certain category and run it only once a day (or so)? What do you think @mrlacey?

**Comment by mrlacey** [comment]
This original issue was only to add VB support to the existing tests so I'll complete this doing that. I've also raised #1807 as a follow-on for adding more sophisticated checking of backward compatibility at a later date.

**Comment by mrlacey** [comment]
just tests so can be closed as now merged into dev branch
